### PR TITLE
[PDI-15140] The trans.getExecutingUser() method is broken with nested…

### PR DESCRIPTION
**Pull request by stanislau-strelchanka**
… "Transformation Executor" calls - pass repository to nested transformation

### CODE REVIEW

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, please review

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java
  engine/test-src/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java

Unit test coverage change:
============
org.pentaho.di.job.entries.sql.JobEntrySQL:
 * Instruction Change : **-1.64%**
 * Line Change : **-1.94%**

org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * Instruction Change : **-.08%**

org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta:
 * Instruction Change :  + .18%

org.pentaho.di.trans.steps.transexecutor.TransExecutor:
 * Branch Change :  + .96%
 * Complexity Change :  + 1.41%
 * Instruction Change :  + 5.25%
 * Line Change :  + 5.26%
 * Method Change :  + 5.26%

org.pentaho.di.trans.steps.transexecutor.TransExecutorParameters:
 * Complexity Change :  + 5.26%
 * Instruction Change :  + .88%
 * Line Change :  + 1.49%
 * Method Change :  + 6.67%

org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * Instruction Change : **-.18%**

org.pentaho.di.trans.Trans:
 * Branch Change :  + .91%
 * Complexity Change :  + 1.55%
 * Instruction Change :  + 1.64%
 * Line Change :  + 1.94%
 * Method Change :  + 4.31%

Integration test coverage change:
============
org.pentaho.di.trans.step.BaseStep:
 * Complexity Change : **-.18%**
 * Instruction Change : **-.20%**
 * Line Change : **-.38%**
 * Method Change : **-.49%**

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, please review

File: engine/test-src/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java [hunk 1]
```diff
@@ -93,6 +96,33 @@ public void tearDown() {
...
+    when( transMetaMock.listVariables() ).thenReturn( new String[0] );
+    when( transMetaMock.listParameters() ).thenReturn( new String[0] );
+    when( transExecutorDataMock.getExecutorTransMeta() ).thenReturn( transMetaMock );
+
+    Trans internalTrans = executor.createInternalTrans();
+    assertNotNull( internalTrans );
+
+    Trans parentTrans = internalTrans.getParentTrans();
+    assertEquals( parentTrans, transParentMock );
+    assertEquals( parentTrans.getRepository(), repositoryMock );
```
Comment (mchen-len-son) [inline engine/test-src/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java]: I think here should be internalTrans.getRepository() check

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, updated, please review

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/src/org/pentaho/di/trans/steps/transexecutor/TransExecutor.java
  engine/test-src/org/pentaho/di/trans/steps/transexecutor/TransExecutorUnitTest.java

Unit test coverage change:
============
org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory:
 * Branch Change :  + 7.14%
 * Complexity Change :  + 7.69%

org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * Instruction Change :  + .08%

org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta:
 * Instruction Change : **-.07%**

org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta:
 * Instruction Change :  + .18%

org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * Branch Change :  + 3.33%
 * Complexity Change :  + 1.85%
 * Instruction Change :  + .28%

org.pentaho.di.trans.steps.transexecutor.TransExecutor:
 * Branch Change :  + .96%
 * Complexity Change :  + 1.41%
 * Instruction Change :  + 5.25%
 * Line Change :  + 5.26%
 * Method Change :  + 5.26%

org.pentaho.di.trans.steps.transexecutor.TransExecutorParameters:
 * Complexity Change :  + 5.26%
 * Instruction Change :  + .88%
 * Line Change :  + 1.49%
 * Method Change :  + 6.67%

org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * Instruction Change : **-.18%**

org.pentaho.di.trans.Trans:
 * Branch Change :  + 1.01%
 * Complexity Change :  + 1.69%
 * Instruction Change :  + 1.71%
 * Line Change :  + 2.05%
 * Method Change :  + 4.78%

**Pull request merged**